vendors/nordic/boards/nrf52840-dk/ports/common_io/iot_flash.c

```diff
@@ -0,0 +1,490 @@
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ *
+ * http://aws.amazon.com/freertos
+ * http://www.FreeRTOS.org
+ */
+
+/**
+ * @file iot_flash.c
+ * @brief HAL Flash implementation on NRF52840 Development Kit
+ */
+
+#include "FreeRTOS.h"
+#include "task.h"
+
+/* Nordic Board includes. */
+#ifdef SOFTDEVICE_PRESENT
+    #include "nrf_soc.h"
+    #include "nrf_sdh.h"
+    #include "nrf_sdm.h"
+    #include "nrf_mbr.h"
+    #include "nrf_sdh_ble.h"
+    #include "nrf_fstorage_sd.h"
+#else
+    #include "nrf_drv_clock.h"
+    #include "nrf_fstorage_nvmc.h"
+#endif
+#include "nrf52840_peripherals.h"
+
+/* Main includes. */
+#include "iot_config.h"
+#include "iot_flash.h"
+#include "iot_flash_config.h"
+
+#ifdef IOT_LOG_LEVEL_GLOBAL
+    #define LIBRARY_LOG_LEVEL    IOT_LOG_LEVEL_GLOBAL
+#else
+    #define LIBRARY_LOG_LEVEL    IOT_LOG_LEVEL_IOT_FLASH
+#endif
+
+#define LIBRARY_LOG_NAME         ( "IOT_FLASH" )
+#include "iot_logging_setup.h"
+
+
+/* Flash constants */
+#define FLASH_PAGE_SIZE         0x1000     /* 4k bytes */
+#define FLASH_SECTOR_SIZE       0x1000     /* 4k bytes */
+#define FLASH_BLOCK_SIZE        0x1000     /* 4k bytes */
+#define FLASH_SECTOR_MASK_4K    0x0FFF     /* 4k mask */
+
+#define MAX_WR_SIZE             NRF_FSTORAGE_SD_MAX_WRITE_SIZE
+#define FLASH_PADDING_BYTE      0xFF       /* NAND-flash default logic value */
+
+static void prvFlashEventHandler( nrf_fstorage_evt_t * p_evt );
+
+NRF_FSTORAGE_DEF( nrf_fstorage_t iot_fstorage ) =
+{
+    /* Set a handler for fstorage events. */
+    .evt_handler = prvFlashEventHandler,
+};
+
+typedef enum
+{
+    IOT_FLASH_CLOSED = 0u,
+    IOT_FLASH_OPENED = 1u
+} IotFlashState_t;
+
+/* @brief flash context */
+typedef struct IotFlashDescriptor
+{
+    IotFlashInfo_t xFlashInfo;    /* flash info structure */
+    IotFlashCallback_t xCallback; /* callback */
+    void * pvUserContext;         /* user context to provide in callback */
+    uint8_t ucState;
+} IotFlashDescriptor_t;
+
+#define DEFAULT_FLASH_DESC                          \
```

Comment:
Is DEFAULT_FLASH_DESC used anywhere besides line 112?

Reply:
If you search in the file you'll see all the usages. It's used at descriptor open and close